crates/biome_deserialize/src/diagnostics.rs

```diff
@@ -54,6 +54,18 @@ impl DeserializationDiagnostic {
         .with_range(range)
     }
 
+    /// Emitted when a generic node has an incorrect type
+    pub fn new_out_of_bound_number(
+        min: impl Display,
+        max: impl Display,
+        range: impl AsSpan,
+    ) -> Self {
+        Self::new(markup! {
+            "The number should be between "<Emphasis>{min}</Emphasis>" and "<Emphasis>{max}</Emphasis>"."
```

Comment:
Hum, I wonder: should we pass a string to `visit_u64`, `visit_i64` and `visit_f64` instead of a numeric type? This could allow arbitrary precision integers / floats.

Reply:
Good call, I think we should